### PR TITLE
Replace ReactPerf with new implementation

### PR DESCRIPTION
**Pull request by gaearon**
This removes the old ReactPerf implementation and replaces it with the one from #6046.
Corresponding React Native PR: https://github.com/facebook/react-native/pull/7283

To be exact:

* The old `ReactPerf`, `ReactDefaultPerf`, and `ReactPerfAnalysis` are removed.
* `ReactDebugTool` (added in #6046) takes care of measurements, and `ReactPerfAnalysis` is renamed to just `ReactPerf`.

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
@gaearon updated the pull request.

File: src/renderers/dom/shared/ReactDefaultInjection.js [hunk 1]
```diff
@@ -91,14 +90,6 @@ function inject() {
   );
 
   ReactInjection.Component.injectEnvironment(ReactComponentBrowserEnvironment);
-
-  if (__DEV__) {
```
Comment (gaearon) [inline src/renderers/dom/shared/ReactDefaultInjection.js]: This block has moved to `ReactDebugTool`.

**Comment by facebook-github-bot-3** [comment]
@gaearon updated the pull request.

**Comment by facebook-github-bot-7** [comment]
@gaearon updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@gaearon updated the pull request.

**Comment by gaearon** [comment]
@sebmarkbage 

This is ready for review now, and has tests salvaged from the previous implementation.
I tested this with https://github.com/facebook/react-native/pull/7283, and it works well.

**Comment by gaearon** [comment]
Rebased on the current master.
@sebmarkbage Can you please take a look until this gets merge conflicts again? 😄

**Comment by facebook-github-bot-8** [comment]
@gaearon updated the pull request.

**Comment by sebmarkbage** [comment]
What's the summary of the current state of the new perf? Where have it been deployed? As anyone other than us tested it?

Seems good to at least have some eyes on it before we kill the old one forever, no?

**Comment by gaearon** [comment]
I tested it on some of the open source apps/examples that I maintain. The results are consistent with what I’d expect them to be. Internally we don’t really use the console version of ReactPerf so I’m not sure what you mean by having it deployed. 

>Has anyone other than us tested it?

No, but we can cut 15.1.0-alpha and ask people to try.

>Seems good to at least have some eyes on it before we kill the old one forever, no?

Sure, I agree. I still think it’s safe to land in master because if we find bugs, we can add more test cases for them. I don’t think we’ll find bugs though until we release this in some form.

**Comment by gaearon** [comment]
Ping @sebmarkbage, anything I can do on my side?

**Comment by sebmarkbage** [comment]
Ok. Hope this works! :)

**Comment by gaearon** [comment]
Let’s cut 15.1.0-alpha with this? I have some people in mind who might help us test it!
cc @slorber

**Comment by slorber** [comment]
I'd like to @gaearon but won't be there for the next 2 weeks :)

**Pull request merged**